### DISCUSSION TITLE
Sharing User Resources between 'Admin' and 'Owner' Roles in FilamentPHP

### DISCUSSION THREAD

**Discussion by ghost** [filamentphp/filament#10391] [Help]
I'm working on a FilamentPHP-based admin panel with separate dashboards for 'Admin' and 'Owner' roles. I've successfully created separate resources for each role using namespaces (e.g., App\Filament\Resource\Admin and App\Filament\Resource\Owner) with separate admin and owner panels. However, I'm facing a challenge with the 'User' resource, as I want it to be shared between both roles.

Creating a new 'User' resource for 'Owner' feels repetitive, and I'm looking for a more efficient solution. Is there a way to share the 'User' resource between 'Admin' and 'Owner' without duplicating code? Additionally, any suggestions on how to improve my current approach are welcome, as I'm feeling uncertain about my implementation.

---

I am trying to create two separate logins for Admin and Owner users, each with different permissions. I heard that I could use policy to implement this so that I don't have to create resources separately. However, I'm not sure if I can still use two panels for admin and owner users to have two separate logins (URLs). Can you please help me with the implementation? Thank you in advance.

**Comment by leandrocfe** [comment]
There are `discoverResources()` `discoverPages()` in the panel provider. You could create a shared folder and use it in both

```php
->discoverResources(in: app_path('Filament/Shared/Resources'), for: 'App\\Filament\\Shared\\Resources')
```
You could also create a "extended resource" like this:

```php
namespace App\Filament\Owner\Resources;

use App\Filament\Admin\Resources\UserResource as BaseUserResource;

class UserResource extends BaseUserResource
{
}

```

**Reply by ghost** [reply]
Imagine you have two panels named `Admin` and `Owner` at `App\Filament\Resource\Admin` and `App\Filament\Resource\Owner`. As you suggested earlier, one can create a folder called `Sharded` to place all the shared resources. To use the shared resources and its own resources, one can change the `discoverResources()` and `discoverPages()` to discover a path in the panel provider of `Owner`. Therefore, it is possible to use both `App\Filament\Resource\Owner` and `App\Filament\Resource\Shared`. Thank you.

**Reply by ghost** [reply]
I need to build an admin panel that has **two logins**: one for the admin and another for the owners. The admin should be able to do all the CRUD operations, while the owner should only be able to view the user. Additionally, the admin should only be able to view products, while the owner should be able to do all the CRUD operations. I am currently using policy and spatie Laravel permission for role-based access control. I would appreciate any guidance on how to achieve this kind of functionality. Thank you in advance.

**Reply by EbenezerOyenuga** [reply]
In the getPages() function you return the array of pages based on the user role that was authenticated
Also you might have to remove the Resource from the Shared Resources and create a new page for viewing for the admin
